### ISSUE TITLE
Bounding box vs instance segmentation 

### ISSUE THREAD

**Issue by OAT7963** [ultralytics/yolov5#1123]
## ❔Question
Hi, should I use bounding boxes or instance segmentation to label my training data if I want to use the model to detect and classify vehicles in case of traffic congestion. Which would yield better result?

Below is an traffic congestion image example. 

Thanks
![image](https://user-images.githubusercontent.com/58838171/95736606-06f2f180-0cb9-11eb-88b6-1164084b5eb5.png)

**Comment by glenn-jocher** [comment]
@OAT7963 the best way to find out is to try both and then work from empirical results.

**Comment by whoafridi** [comment]
I would say both are (bounding boxes or instance segmentation) different set of problems where for boxes you use rectangle for data labels & segmentation use polygon. In addition both have different format of annotations also. So, the best way to find out is to try both and then work from empirical results. Possibly segmentation looks great for traffic congestion problem instead of bbox .

**Comment by glenn-jocher** [comment]
We did a traffic demo earlier here: https://www.youtube.com/watch?v=iUpZagGyhlM&ab_channel=UltralyticsLLC

Though of course the angle of the camera will impact results, among other things.

**Comment by MikeHatchi** [comment]
Hi Everyone,

Then, I've probably missed something.
Yolov5 asks for training: class x_center y_center width height, not any color_class (or color_category) data thus @OAT7963 needs to add this input parameter from scratch within the model, doesn't he?

**Comment by glenn-jocher** [comment]
@MikeHatchi yes for @OAT7963 to train segmentation tasks significant feature additions would be required to the YOLOv5 repository (loss function, dataloader, testing methodology), in addition to model architectural changes (a head suitable for segmentation outputs).

The COCO dataset itself already supports segmentation training as it provides segmentation labels.

**Comment by MikeHatchi** [comment]
@glenn-jocher Thx!!

FYI, I've made this little pull request: https://github.com/vrona/yolov5/pull/1, as if not it raised "Key Error" when lauching training.

**Comment by glenn-jocher** [comment]
@MikeHatchi ah, yes this was a recent breaking change in master, it’s mentioned in the 3.1 release notes.

Note that you submitted a PR to a fork though rather than here.
